core/src/main/java/jenkins/util/VirtualFile.java

```diff
@@ -327,11 +327,9 @@ public Collection<String> call() throws IOException {
 
         private static void collectFiles(VirtualFile d, Collection<String> names, String prefix) throws IOException {
             for (VirtualFile child : d.list()) {
-                if (child.isFile()) {
-                    names.add(prefix + child.getName());
-                } else if (child.isDirectory()) {
-                    collectFiles(child, names, prefix + child.getName() + "/");
-                }
+                // JENKINS-73837 - Empty Directories are now included in the list
```

Comment:
please don't include point in time comments especially about something being fixed
```suggestion
```